### ISSUE TITLE
Migrate branchReset integration test to new pattern

### ISSUE THREAD

**Issue by jesseduffield** [jesseduffield/lazygit#2214]
Now that we have a new integration test pattern in place we want to migrate each of our old tests to the new pattern.

* [guide to integration tests](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) (including migration strategy)
* [blog post](https://jesseduffield.com/IntegrationTests/)

the branchReset integration test lives at `test/integration/branchReset`

If you want to pick up this issue and have any questions, let me know!

**Comment by con-mark** [comment]
Hi @jesseduffield  I can have a look although not great with go just yet so feel free to give to someone else :)

**Comment by con-mark** [comment]
Hi @jesseduffield ,

I just want to confirm I should be testing the same functionality as described in
https://github.com/jesseduffield/lazygit/blob/master/test/integration/branchReset/test.json

Or is the test for a reset master?

If it is following the json I guess I can use the following for guidance?
https://github.com/jesseduffield/lazygit/tree/master/test/integration_new/branch/rebase/expected/repo
https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/tests/branch/rebase.go

Just trying to get my head around the structure of the tests at the moment :)

**Comment by jesseduffield** [comment]
yep that's right, you want to capture the same behaviour as in the original test. And you can run that test and watch it play by running the following:
```go
SPEED=1 go run pkg/integration/deprecated/cmd/runner/main.go branchReset
```

And yep that rebase.go file is a good one for guidance.

**Comment by jesseduffield** [comment]
All good @con-mark :)
